### ISSUE TITLE
fish shell segfaults trying to run script in a NixOS docker

### ISSUE THREAD

**Issue by locallycompact** [NixOS/nixpkgs#59357]
Hi, I'm trying to execute a fish script which runs fine on my local machine on NixOS, but fails in a NixOS docker. fish --version works in the docker and returns 3.0.2, but attempting to run a script results in many lines of the form.

```
<4> fish: Set status of job 2194 (test ...) to 0 using short circuit
<3> fish: Created job 2194 from command 'test -f /etc/fish/config.fish' with pgrp -2
<4> fish: Start job 2194, gid -2 (test -f /etc/fish/config.fish), COMPLETED, NON-INTERACTIVE
<5> fish: intern /etc/fish/config.fish
<5> fish: intern /etc/fish/config.fish
<5> fish: intern /etc/fish/config.fish
<5> fish: intern /etc/fish/config.fish
<5> fish: intern /etc/fish/config.fish
<4> fish: Skipping fork: no output for internal builtin 'test'
```

Full Log here:

https://gitlab.com/zenhaskell/infra-dockers/pipelines/56255895

Any script will fail with a segfault. Something as simple as

```
#!/usr/bin/env fish
echo "foo"
```

will fail.

**Comment by matthewbauer** [comment]
The fact that it segfaults sounds liker a fish bug to me. With bash, I get this:

```
# test.sh file
source ./test.sh
```

```
$ source ./test.sh
./test.sh:1: job table full or recursion limit exceeded
```

**Comment by locallycompact** [comment]
I don't really understand what you've said. What are you comparing with bash?

**Comment by faho** [comment]
>The fact that it segfaults sounds liker a fish bug to me

@matthewbauer: The important part is that there's an infinite recursion here, which is bogus and needs to be removed.

How fish handles this form of infinite recursion is a secondary issue - even if it just errored, it would still be an error and should still be fixed in the nix package.

**Comment by xaverdh** [comment]
I checked that this resolves #34656, did not check the docker case, but it should also fix that.
